### PR TITLE
✨ Support querying by keyword

### PR DESCRIPTION
**Pull request by williamchong**
@lancatlin

### CODE REVIEW

File: store/iscn.ts [hunk 1]
```diff
@@ -161,6 +163,25 @@ export default class ISCN extends VuexModule {
...
 
+  @Action
+  async queryISCNByField({
+    keyword,
+  }: { keyword: string }): Promise<ISCNRecordWithID[]> {
+    this.context.commit('clearRecords');
+    let result: ISCNRecord[] = ([] as ISCNRecord[])
+    try {
+      this.context.commit('setIsLoading', true)
+      const { data } = await axios.get(`${network.apiURL}/iscn/records?keyword=${encodeURIComponent(keyword)}`)
```
Comment (lancatlin) [inline store/iscn.ts:174]: I think it should be `/iscn/keywords?q=` for search all function (includes iscn id, owner, stakeholder, keywords). `keyword` will only search `contentMetadata.keywords`.

File: pages/index.vue [hunk 2]
```diff
@@ -116,7 +116,12 @@ import { logTrackerEvent } from '~/utils/logger'
 
 const iscnModule = namespace('iscn')
 
-@Component
+@Component({
+  fetch({ redirect, query }) {
+    const { q } = query;
+    if (q) redirect({ name: 'search-keyword___en', query });
```
Comment (nwingt) [inline pages/index.vue:122]: can use `localeRoute({ name: 'search-keyword' })` ?

File: store/iscn.ts (hunk 1 above)
Comment (williamchong) [reply store/iscn.ts:174]: yes, the requirement was to specifically query by `keyword`. I think `queryISCNByKeyword`(poor name btw) above should be using `q` for search-all, and this function is for specific fields
https://github.com/likecoin/app-like-co/pull/175 should replace the above function by `q` after we fix performance issue (I based this PR on #175 actually)

File: pages/index.vue (hunk 2 above)
Comment (williamchong) [reply pages/index.vue:122]: typescript doesn't have `localeLocation` in context, how can I hack it..

File: tsconfig.json [hunk 3]
```diff
@@ -27,11 +27,11 @@
       ]
     },
     "types": [
+      "@types/node",
       "@nuxt/types",
+      "@nuxtjs/sentry",
```
Comment (nwingt) [inline tsconfig.json:32]: Why changing the order?

Comment (williamchong) [reply tsconfig.json:32]: it seems that the type below (axios, i18n) extends the type above (node => nuxt) so the order some how matters

**Review by nwingt — APPROVED** [review]

**Pull request merged**